recipes/beyla/rbac.yaml

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: otel-collector
+rules:
+- apiGroups: [""]
+  resources: ["pods", "namespaces", "nodes"]
+  verbs: ["get", "watch", "list"]
+- apiGroups: ["apps"]
+  resources: ["replicasets"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["extensions"]
+  resources: ["replicasets"]
+  verbs: ["get", "list", "watch"]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: otel-collector
+subjects:
+- kind: ServiceAccount
+  name: otel-collector
+  namespace: default
```

Comment:
is the collector service account created in the `default` namespace when you create a collector config, or the current namespace? Wondering if I deployed my collector (and serviceaccount?) in namespace `foobar`, will this rolebinding still work

Reply:
Wdym by "current namespace"?

I think it has to match the collector's SA namespace based on [the k8sattributes docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.90.0/processor/k8sattributesprocessor/README.md#role-based-access-control)

Reply:
I mean like if I run `kubectl apply -f collector-config.yaml -n foobar` to deploy the collector in that ns, then I run `kubectl apply -f rbac.yaml`, will it work? I'm wondering where the operator creates the collector ServiceAccount when it deploys a collector... if that's in `default`, then this rolebinding will work. But if it's in `foobar`, then I think this would be broken

Reply:
Just tried it out and the operator creates the `otel-collector` SA in the namespace you deploy the collector in automatically. So they have to match.

Reply:
ok, we should call that out then

Reply:
Added a note to the README